core/html_api.php

```diff
@@ -1390,10 +1390,22 @@ function html_status_legend( $p_display_position, $p_restrict_by_filter = false
 	$t_workflow = config_get( 'status_enum_workflow' );
 	if( !empty( $t_workflow ) ) {
 		foreach( $t_status_array as $t_status => $t_name ) {
+			# Check if the workflow allows leaving the state
 			if( !isset( $t_workflow[$t_status] ) ) {
 
+				# Check if the workflow allows entering the state
+				$t_can_enter = false;
+				foreach( $t_workflow as $t_values ) {
+					if( MantisEnum::hasValue( $t_values, $t_status ) ) {
+						$t_can_enter = true;
+						break;
+					}
+				}
+
 				# drop elements that are not in the workflow
-				unset( $t_status_array[$t_status] );
+				if( !$t_can_enter ) {
+					unset( $t_status_array[$t_status] );
```

Comment:
It is generally not recommended to change an array within a foreach on the array.  In some languages (e.g. C#) this is banned.  Not sure about PHP, but you may consider doing this outside the loop.